File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/detail/ui/TransactionDetailPresenterModule.java
package com.fintechplatform.ui.transactions.ui.detail.ui;

import com.fintechplatform.ui.transactions.models.TransactionDetailHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class TransactionDetailPresenterModule {

    private TransactionDetailContract.View view;

    public TransactionDetailPresenterModule(TransactionDetailContract.View view) {
        this.view = view;
    }

    @Provides
    @Singleton
    TransactionDetailContract.Presenter providesTransactionDetailPresenter(TransactionDetailHelper helper) {
        return new TransactionDetailPresenter(view, helper);
    }

    @Provides
    @Singleton
    TransactionDetailHelper providesTransactionDetailHelper() {
        return new TransactionDetailHelper();
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/ui/detail/di/TransactionDetailPresenterModule.kt
package com.fintechplatform.ui.transactions.ui.detail.di

import com.fintechplatform.ui.transactions.models.TransactionDetailHelper
import com.fintechplatform.ui.transactions.ui.detail.TransactionDetailContract
import com.fintechplatform.ui.transactions.ui.detail.TransactionDetailPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class TransactionDetailPresenterModule(private val view: TransactionDetailContract.View) {

    @Provides
    @Singleton
    internal fun providesTransactionDetailPresenter(helper: TransactionDetailHelper): TransactionDetailContract.Presenter {
        return TransactionDetailPresenter(view, helper)
    }

    @Provides
    @Singleton
    internal fun providesTransactionDetailHelper(): TransactionDetailHelper {
        return TransactionDetailHelper()
    }
}
